integrations/oci/tls-certificates/src/main/java/io/helidon/integrations/oci/tls/certificates/DefaultOciCertificatesTlsManager.java

```diff
@@ -191,19 +178,14 @@ boolean loadContext() {
                 throw new RuntimeException("Unable to find X.509 trust manager in download");
             }
 
-            // establish the new context
-            // context.init(kmf.getKeyManagers(), tmf.getTrustManagers(), secureRandom);
-            keyManager(keyManager.get());
-            trustManager(trustManager.get());
-
+            if (initialLoad) {
+                initSslContext(tlsConfig, secureRandom, kmf.getKeyManagers(), tmf.getTrustManagers());
+            } else {
+                reload(keyManager, trustManager);
+            }
             return true;
-        } catch (RuntimeException
-                 | KeyStoreException
-                 | NoSuchAlgorithmException
-                 | IOException
-                 | CertificateException
-                 | UnrecoverableKeyException e) {
-            throw new RuntimeException("Error while loading context from OCI", e);
+        } catch (KeyStoreException e) {
```

Comment:
Seems we need a broader exception catch here since we are also downloading the pk from OCI which will throw a RuntimeException and not a KeyStoreException.

Reply:
1. you should not catch "any" runtime exception - some of the code throws our custom exception
2. you should never throw a `RuntimeException` type - we should either use a custom exception, or one of the predefined ones that make sense (this is part of our development guidelines)

If you want to wrap runtime exceptions from OCI (not sure this is really needed):
1. create a custom exception in our TLS module to mark our exceptions (when we fail to create secure random, keystore, factories etc.)
2. catch this exception and re-throw it without wrapping
3. catch a runtime exception and wrap it in your exception